cmd/mirror-main.go

```diff
@@ -493,25 +488,15 @@ exit:
 
 // when using a struct for copying, we could save a lot of passing of variables
 func (mj *mirrorJob) mirror(ctx context.Context, cancelMirror context.CancelFunc) {
-	if globalQuiet || globalJSON {
-	} else {
-		// Enable progress bar reader only during default mode
-		mj.scanBar = scanBarFactory()
-	}
-
 	// start the status go routine
 	mj.startStatus()
 
-	// Starts additional watcher thread for watching for new events.
-	if mj.isWatch {
-		go mj.watchMirror(ctx, cancelMirror)
-	}
-
 	// Start mirroring.
 	mj.startMirror(ctx, cancelMirror)
 
+	// Starts watcher loop for watching for new events.
 	if mj.isWatch {
-		<-mj.trapCh
+		mj.watchMirror(ctx, cancelMirror)
```

Comment:
We need to check the error message returned by this error, and I guess only print it as a warning.

Reply:
Requires some restructuring let me do that.